source/grid/grid_tools.cc

```diff
@@ -5088,12 +5088,16 @@ namespace GridTools
     MPI_Comm                                  mpi_communicator)
   {
 #ifndef DEAL_II_WITH_MPI
-    (void)local_description;
-    (void)mpi_communicator;
-    Assert(false,
-           ExcMessage(
-             "GridTools::build_global_description_tree() requires MPI."));
-    return RTree<std::pair<BoundingBox<spacedim>, unsigned int>>{};
+    // Building a tree with the only boxes available without MPI
+    std::vector<std::pair<BoundingBox<spacedim>, unsigned int>> boxes_index(
+      local_description.size());
+    for (const auto &box : local_description)
```

Comment:
You probably don't need this loop at all. In the inner loop you are setting all the entries already.

Reply:
Ha, yes -- if the loop body doesn't reference the loop variable, then there's probably a bug somewhere :-) Good catch!